core/src/main/java/com/graphhopper/storage/BaseGraph.java

```diff
@@ -958,17 +949,26 @@ public int getReverseEdgeKey() {
         }
 
         @Override
-        public String getName() {
-            int nameRef = store.getNameRef(edgePointer);
-            String name = (String) baseGraph.edgeKVStorage.get(nameRef, EDGEKV_NAME_KEY);
-            // preserve backward compatibility (returns null if not explicitly set)
-            return name == null ? "" : name;
+        public EdgeIteratorState setKeyValues(Map<String, Object> map) {
+            int kvEntryRef = (int) baseGraph.edgeKVStorage.add(map);
+            if (kvEntryRef < 0)
+                throw new IllegalStateException("Too many maps are stored, currently limited to 'int' pointer");
```

Comment:
I meant theoretically `baseGraph.edgeKVStorage` could return a long like `long x = 6_000_000_000L`. Then `(int)x` would be > 0 but it would still be an overflow. So should it not be something like:

```java
long ptr = baseGraph.edgeKVStorage.add(map);
if (ptr > Integer.MAX_VALUE)
   throw new IllegalStateException("EdgeKVStorage overflow, the number of stored key value pairs exceeds our four byte pointer limitation");
 if (ptr < 0)
    throw new IllegalStateException("Negative pointer returned from EdgeKVStorage#add");
 int kvEntryRef = (int) ptr;     
```

Reply:
Yes, the added map cannot have more than `Integer.MAX_VALUE` entries, but the number returned from the `add` method is the byte pointer and this can be increased by more than `Integer.MAX_VALUE`, e.g. when the map values are large (and numerous), doesn't it?

Reply:
Ah ok because the number of key value pairs and the maximum string length are limited. Ok yes this was only a theoretical issue and apparently it is not even possible to run into a problem. Don't know. When I cast to a long the obvious thing to do seems to be checking if the value is in the sane range where longs are converted to int without any overflow before doing the cast, but if you rather check if the number after the cast is negative I won't complain :)